### PR TITLE
Add contract generator

### PR DESCRIPTION
**Pull request by kannonfodder**
Added in contract generator class which acts like Pact Builder
Currently generates contracts for all registered requests in the mock provider service when cleaning up
included ConsumerApi sample (with sample tests)

### CODE REVIEW

File: src/ConcordNet/ConcordGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            return this;
+        }
+
+        public IConcordGenerator HasContractWith(string providerService)
+        {
+            Provider = providerService;
+            return this;
+        }
+
+        [Obsolete]
```
Comment (morganlaneap) [inline src/ConcordNet/ConcordGenerator.cs]: What is the reason for this being obsolete? Could you put a reason on this?

File: src/ConcordNet/ConcordGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                throw new Exception("Define the service");
+            }
+
+            if (string.IsNullOrEmpty(Provider))
+            {
+                throw new Exception("Define the provider");
+            }
+
+            var contracts = _providerService.GetContracts();
+            File.WriteAllText($"{contractDirectory}{Consumer}-{Provider}.json", JsonSerializer.Serialize(contracts));
```
Comment (morganlaneap) [inline src/ConcordNet/ConcordGenerator.cs]: Would it be possible to pass in a `JsonSerializerSettings` that indents the output JSON file for readability. PACT outputs them in a readable format.

**Review by morganlaneap — CHANGES_REQUESTED** [review]

File: src/ConcordNet/ConcordGenerator.cs (hunk 1 above)
Comment (kannonfodder) [reply src/ConcordNet/ConcordGenerator.cs]: Ohh sure,  this was so we could do minimal changes in the APIs but would get compiler warnings

File: src/ConcordNet/ConcordGenerator.cs (hunk 2 above)
Comment (kannonfodder) [reply src/ConcordNet/ConcordGenerator.cs]: Great idea

File: src/ConcordNet/ConcordGenerator.cs (hunk 1 above)
Comment (kannonfodder) [reply src/ConcordNet/ConcordGenerator.cs]: Honestly I'll just remove it

**Review by morganlaneap — APPROVED** [review]

**Pull request merged**